### PR TITLE
bug fixes for Hydjets and Pyquen

### PR DESCRIPTION
**Pull request by wouf**
#### PR description:
This is bugfix for cms-sw/cmssw#36316 . It fixes some bugs from [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cae5ac/22136/llvm-analysis/) 
Also two cff files unused in Configuration/Generator have been deleted in this PR.

#### PR validation:
Hydjet2 - wf 160
Hydjet - wf 150
Pyquen - wf 300

### CODE REVIEW

**Comment by SiewYan** [comment]
please test

**Comment by qliphy** [comment]
type bugfix

**Comment by perrotta** [comment]
@wouf could you please write a title slightly more descriptive for this PR? At least Hydjet should be present in it

**Comment by wouf** [comment]
Please hold it, the solution is not correct!

File: GeneratorInterface/Hydjet2Interface/src/Hydjet2Hadronizer.cc [hunk 1]
```diff
@@ -462,6 +459,9 @@ bool Hydjet2Hadronizer::get_particles(HepMC::GenEvent *evt) {
     }
     ihy++;
   }
+  if (sub_vertices)
```
Comment (Dr15Jones) [inline GeneratorInterface/Hydjet2Interface/src/Hydjet2Hadronizer.cc:462]: There is no need to check that the pointer is non-null before calling delete. Internally delete does such a check.

**Comment by perrotta** [comment]
urgent

**Comment by perrotta** [comment]
> Please hold it, the solution is not correct!

Please @wouf proceed with the correct solution, taking also into account the [suggestion](https://github.com/cms-sw/cmssw/pull/36918#pullrequestreview-880231488) from Chris: wf 148.0 keeps crashing in the IBs, and we woukdn't like to propagate that crash to pre5.
An acceptable (intermediate) solution could be to strip the bug fixes specific to solve the issue with that wf in a separate PR, and take your time to proceed with the other issues addressed by this PR

**Comment by perrotta** [comment]
> @perrotta indeed issue with memory leak is more complicated than I thought before, I will remove wrong solution from this PR

(probably a smart pointer could be used, instead)

**Comment by wouf** [comment]
> > Please hold it, the solution is not correct!
> 
> Please @wouf proceed with the correct solution, taking also into account the [suggestion](https://github.com/cms-sw/cmssw/pull/36918#pullrequestreview-880231488) from Chris: wf 148.0 keeps crashing in the IBs, and we woukdn't like to propagate that crash to pre5. An acceptable (intermediate) solution could be to strip the bug fixes specific to solve the issue with that wf in a separate PR, and take your time to proceed with the other issues addressed by this PR

@perrotta This PR can be merged, I will put memory leak bugfix to other.

**Comment by perrotta** [comment]
@cms-sw/generators-l2 please have a look: this is needed in pre5

**Comment by SiewYan** [comment]
@wouf , thank for your effort. If I understood correctly, you have removed the two unused cfi and corrected the smart pointer. The bugfix PR will come later?

**Comment by perrotta** [comment]
> @wouf , thank for your effort. If I understood correctly, you have removed the two unused cfi and the pointer. The bugfix PR will come later?

@SiewYan this is THE bugfix for the crash that we observe in wf 148 in the IB, due to an `int` parameter being declared as a `bool` in a few configurations

Then, there is some possible memory leak reported by the static analyzer, and that's what differed to a future PR (unless the `delete` statements implemented here already solve it: I'm realizing that the bot tests were not run yet, doing it now)

**Comment by perrotta** [comment]
please test workflow 148.0

**Comment by SiewYan** [comment]
@perrotta , thank you for the description, it is very hard to follow from the long thread... I will sign it once the test is a success.

**Comment by SiewYan** [comment]
+1

**Comment by perrotta** [comment]
+1
